### ISSUE TITLE
[release-1.21] Cluster fail to provision when using AWS cloud provider 

### ISSUE THREAD

**Issue by galal-hussein** [rancher/rke2#1618]
**Environmental Info:**
RKE2 Version: v1.21.3+rke2r1

Node(s) CPU architecture, OS, and Version: 
Ubuntu 20.04

Cluster Configuration:
1 Server

**Describe the bug:**
When not passing `--node-name` to rke2 server and using `--cloud-provider-name=aws` , kubelet fail to register itself with kubernetes api because of the NodeRestriction Admission plugin https://github.com/rancher/rancher/issues/34105#issuecomment-897258413
  
**Steps To Reproduce:**
- Create ec2 instance on AWS
- Tag instance, subnet, vpc, and security group with kubernetes tag
- Start rke2 server with option --cloud-provider-name=aws

**Expected behavior:**
The cluster should start normally

**Actual behavior:**
Kubelet fails to register with the following error:

```
Attempting to register node ip-172-28-68-60.eu-west-1.compute.internal
Unable to register node "ip-172-28-68-60.eu-west-1.compute.internal" with API server: nodes "ip-172-28-68-60.eu-west-1.compute.internal" is forbidden: node "ip-172-28-68-60" cannot modify node "ip-172-28-68-60.eu-west-1.compute.internal"
```

**Comment by cjellick** [comment]
I think we want the 1.21.3+rke2r2 milestone

**Comment by brandond** [comment]
@cjellick I thought we were not actually planning on doing a 1.21.3+rke2r2 release - the current RCs are just for QA. The next actual release will be 1.21.4+rke2r1.

**Comment by galal-hussein** [comment]
> @cjellick I thought we were not actually planning on doing a 1.21.3+rke2r2 release - the current RCs are just for QA. The next actual release will be 1.21.4+rke2r1.

Thats what I thought too

**Comment by brandond** [comment]
This has been merged to master; I have updated the milestone to 1.22 and created a backport issue for 1.21.

**Comment by brandond** [comment]
Edit: was merged to release-1.21 first, will convert the other issue into a forwardport

**Comment by rancher-max** [comment]
#### Validated using v1.21.3-rc6+rke2r2

No longer need to supply node-name as hostname when using aws cloud provider. Can still supply it if desired. It will automatically use an equivalent to `hostname -f` when starting rke2 with aws cloud provider now though.
